native/src/main/java/io/ballerina/stdlib/http/transport/contractimpl/listener/HttpServerChannelInitializer.java

```diff
@@ -427,9 +427,11 @@ protected void configurePipeline(ChannelHandlerContext ctx, String protocol) {
         }
 
         @Override
-        public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) {
+        public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws Exception {
             if (ctx != null && ctx.channel().isActive()) {
                 ctx.writeAndFlush(Unpooled.EMPTY_BUFFER).addListener(ChannelFutureListener.CLOSE);
+            } else {
+                super.exceptionCaught(ctx, cause);
```

Comment:
To be safe, better to do a ctx null check here as well like below comment

Reply:
Fixed in https://github.com/ballerina-platform/module-ballerina-http/pull/563/commits/4031fdd1992727272c7e05b1aa0085d1e42bce8a